src/components/form/signUpForm/signUpForm.js

```diff
@@ -0,0 +1,215 @@
+import { useState } from 'react'
+import { useDispatch } from 'react-redux'
+import { NavLink } from 'react-router-dom'
+import { useFormik } from 'formik'
+import * as Yup from 'yup'
+
+import Title from '../../title'
+import Form from '../formStyles/form'
+import Wrapper from '../formStyles/inputWrapper'
+import Input from '../formStyles/input'
+import ErrorText from '../formStyles/errorText'
+import Svg from '../formStyles/passwordSvg'
+import Button from '../formStyles/button'
+import TextWrapper from '../formStyles/textWrapper'
+import Text from '../formStyles/text'
+import Link from '../formStyles/link'
+
+import routes from '../../../routes/routes'
+import authActions from '../../../redux/auth/authActions'
+
+const SignUpForm = () => {
+  const [togglePass1, setTogglePass1] = useState(false)
+  const [togglePass2, setTogglePass2] = useState(false)
+  const dispatch = useDispatch()
+
+  const handleToggle = (e) => {
+    if (e.target.id === 'togglePass1') {
+      setTogglePass1(!togglePass1)
+    }
+
+    if (e.target.id === 'togglePass2') {
+      setTogglePass2(!togglePass2)
+    }
+
+    return
+  }
+
+  const formik = useFormik({
+    initialValues: {
+      firstName: '',
+      lastName: '',
+      email: '',
+      password: '',
+      confirmPass: '',
+    },
+    validationSchema: Yup.object({
+      firstName: Yup.string()
+        .min(2, 'Must be at least 2 characters')
+        .max(15, 'Must be 15 characters or less')
+        .required('Please enter your name')
+        .matches(
+          /^[a-z][a-z '-.,]{1,21}$|^$/i,
+          'Please use only letters for your name'
+        ),
+      lastName: Yup.string()
+        .min(2, 'Must be at least 2 characters')
+        .max(20, 'Must be 20 characters or less')
+        .required('Please enter your last name')
+        .matches(
+          /^[a-z][a-z '-.,]{1,21}$|^$/i,
+          'Please use only letters for your last name'
+        ),
+      email: Yup.string()
+        .email('Invalid email address')
+        .required('Please enter your email address'),
+      password: Yup.string()
+        .required('Please Enter your password')
+        .matches(
+          /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,
+          'Must contain 8 characters,at least one letter and one number'
+        ),
+      confirmPass: Yup.string()
+        .required('Please Enter your password')
+        .oneOf([Yup.ref('password'), null], 'Passwords must match'),
+    }),
+    onSubmit: () => dispatch(authActions.registerSuccess()),
+  })
+
+  return (
+    <>
+      <Form onSubmit={formik.handleSubmit}>
+        <Title text="Sign Up" />
+
+        <div>
```

Comment:
Unused wrapper, remove

Reply:
this div used for css rules (&:nth-child(n)::before and last-child)
I can't remove it.